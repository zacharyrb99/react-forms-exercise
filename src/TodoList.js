import {useState} from 'react';
import NewTodoForm from './NewTodoForm';
import {v4 as uuid} from 'uuid';

const TodoList = () => {
    const INITIAL_STATE = [
        {text: 'test todo'}
    ]

    const [toDos, setToDos] = useState(INITIAL_STATE);

    const addToDo = (newToDo) => {
        setToDos(toDos => [...toDos, {id: uuid(), ...newToDo}]);
    }

    const removeToDo = (id) => {
        setToDos(toDos => toDos.filter(toDo => toDo.id !== id));
    }

    return (
        <div className='App-ToDos'>
            <h2>To-Do List</h2>
            <NewTodoForm add={addToDo}/>
            <ol>
                {toDos.map(({id, text}) => (
                    <li>
                        {text} <button onClick={() => removeToDo(id)}>X</button>
                    </li>
                ))}
            </ol>
        </div>
    )
}

export default TodoList;