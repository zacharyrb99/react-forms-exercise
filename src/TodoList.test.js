import {render, fireEvent} from '@testing-library/react';
import TodoList from './TodoList';

test('TodoList renders without crashing', () => {
    render(<TodoList/>);
});

test('TodoList matches snapshot' , () => {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment).toMatchSnapshot();
});

test('It adds a new to-do', () => {
    const {queryByText, getByLabelText} = render(<TodoList/>);
    const textInput = getByLabelText('To-Do Text:');
    const button = queryByText('Add To-Do');

    expect(queryByText('Test To-Do')).not.toBeInTheDocument();

    fireEvent.change(textInput, {target: {value: 'Test To-Do'} });
    fireEvent.click(button);

    expect(queryByText('Test To-Do')).toBeInTheDocument();
});