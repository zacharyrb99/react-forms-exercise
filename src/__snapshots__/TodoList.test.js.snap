// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches snapshot 1`] = `[Function]`;
